src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4565,21 +4632,43 @@ def _set_log_analytics_workspace_extension(cmd, resource_group_name, vm, vm_name
 # disk encryption set
 def create_disk_encryption_set(
         cmd, client, resource_group_name, disk_encryption_set_name, key_url, source_vault=None, encryption_type=None,
-        location=None, tags=None, no_wait=False, enable_auto_key_rotation=None):
+        location=None, tags=None, no_wait=False, enable_auto_key_rotation=None, federatedClientId=None,
+        mi_system_assigned=None, mi_user_assigned=None):
     from msrestazure.tools import resource_id, is_valid_resource_id
     from azure.cli.core.commands.client_factory import get_subscription_id
     DiskEncryptionSet, EncryptionSetIdentity, KeyForDiskEncryptionSet, SourceVault = cmd.get_models(
         'DiskEncryptionSet', 'EncryptionSetIdentity', 'KeyForDiskEncryptionSet', 'SourceVault')
-    encryption_set_identity = EncryptionSetIdentity(type='SystemAssigned')
+
+    identity_type = 'SystemAssigned'
+    user_assigned_identities = None
+    if cmd.supported_api_version(min_api='2022-03-02', resource_type=ResourceType.MGMT_COMPUTE,
+                                 operation_group='disk_encryption_sets'):
+        identity_type, user_assigned_identities = \
+            _build_identities_info_from_system_user_assigned(cmd, mi_system_assigned, mi_user_assigned)
+
+    encryption_set_identity = EncryptionSetIdentity(type=identity_type)
+    if user_assigned_identities is not None:
+        if not hasattr(encryption_set_identity, 'user_assigned_identities'):
+            raise CLIError('User assigned identities in disk encryption set is not supported')
```

Comment:
Please use specific error type instead of `CLIError`

Reply:
Could you explain why there is such logic?

Reply:
> Could you explain why there is such logic?

If the sdk version and CLI version will be bound for upgrade, this logic can be removed.